types.go

```diff
@@ -350,7 +350,7 @@ type DocumentQuery struct {
 // DocumentsQuery is the request body for list documents method
 type DocumentsQuery struct {
 	Offset int64    `json:"offset,omitempty"`
-	Limit  int64    `json:"limit,omitempty"`
+	Limit  int64   `json:"limit,omitempty"`
```

Comment:
```suggestion
	// Limit sets the numbers of returned documents from the
	// search queries. The limit can be zero for explicitly
	// return nothing. For retrieving all 
	// returned documents, use NoLimit. 
	Limit  int64   `json:"limit,omitempty"`
```

Reply:
I don't think adding a code comment to every single attribute is a good practice. The code is going to look bloated.

Reply:
No one reads the code. But everyone will benefits from having a documentation on their IDE/text-editor without having to go to the documentation. All you need to do is just hover at the struct/field name.

Take `http.Client` documentation for example: https://pkg.go.dev/net/http@go1.20.4#Client -- they have comments as documentation on everything. It helps the users understand what that is. If you look at the source code (https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/net/http/client.go;l=30-107), yes it's a bit messy thanks to lots and lots of documentation. But as the user who uses the library, it's helpful.

Reply:
Great example! Thanks a lot for the reference.